core/src/Kokkos_OpenMP.hpp

```diff
@@ -179,6 +179,7 @@ namespace Experimental {
 template <>
 struct DeviceTypeTraits<OpenMP> {
   static constexpr DeviceType id = DeviceType::OpenMP;
+  static int device_id(const OpenMP&) { return 0; }
```

Comment:
Shouldn't this be `omp_get_default_device()` like for OpenMPTarget?

Reply:
I did a quick check in the standard and I couldn't find in the standard that the CPU will always have `device_id = 0`, that's why I think it's safer to use `omp_get_default_device()`

Reply:
> What I was trying to say is that the device reported by omp_get_default_device is only for target regions 

What do you mean? That function does not work inside a target region. `When called from within a target region the effect of this routine is unspecified.`

Reply:
Yes, but AFAICT a `device` is only used for `target` constructs, see https://splichal.eu/scripts/sphinx/libgomp/_build/html/openmp-environment-variables/ompdefaultdevice.html#omp-default-device. You use it to define a target region, not inside one.

Reply:
Could not find any documentation on the behavior of `omp_get_default_device` when a target device is not present.